### ISSUE TITLE
Splash Screen

### ISSUE THREAD

**Issue by loljoho** [loljoho/gpquery-rdash#14]
__Overview__
Create splash screen for application loading

__Details__
Create a full-width, full-height, loading splash screen that shows when a full page load is occurring, and automatically hides upon completion of the load.

__Task List__
 - [ ] Create `gp-splash` directive in `gpquery.components` module
 - [ ] Add `gp-splash` to `index.html`

__Resources__
 - Blur Admin `#preloader` element
   - [Index](https://github.com/akveo/blur-admin/blob/master/src/index.html#L57)
   - [Stylesheet](https://github.com/akveo/blur-admin/blob/master/src/sass/theme/_preloader.scss)
   - [Configuration](https://github.com/akveo/blur-admin/blob/master/src/app/theme/theme.run.js#L28)
 - Plugin - johnpapa `.page-splash` with `.ng-hide` style
   - [Index](https://github.com/johnpapa/ng-demos/blob/master/modular/src/client/index.html#L31)
   - [`.ng-hide`](https://github.com/johnpapa/ng-demos/blob/master/modular/src/client/index.html#L11)
   - [Stylesheet](https://github.com/johnpapa/ng-demos/blob/master/modular/src/client/content/styles.css#L235)
   - [Plugin](https://github.com/AngularPlus/AngularPlus)
 - Ben Nadel Tutorial
   - [Tutorial](https://www.bennadel.com/blog/2758-creating-a-pre-bootstrap-loading-screen-in-angularjs.htm)
   - [Index](https://github.com/bennadel/JavaScript-Demos/blob/master/demos/pre-bootstrap-loading-screen-angularjs/index.htm)
```js
function preloaderDirective($animate, $timeout, $q) {

  var directive = {
    restrict: 'EAC',
    template: '<div class="preloader-progress">'
            + '<div class="preloader-progress-bar" ng-style="{width: loadCounter + \'%\'}">'
            + '</div>'
            + '</div>',
    link: link
  };
  return directive;

  function link(scope, el) {
    var counter = 0;
    var timeout;
    
    scope.loadCounter = 0;
    angular.element('body').css('overflow', 'hidden');
    el.addClass('preloader');

    appReady().then(endCounter);
    timeout = $timeout(startCounter);

    function startCounter() {
      var remaining = 100 - counter;
      counter = counter + (0.015 * Math.pow(1 - Math.sqrt(remaining), 2));
      scope.loadCounter = parseInt(counter, 10);
      timeout = $timeout(startCounter, 20);
    }

    function endCounter() {
      $timeout.cancel(timeout);
      scope.loadCounter = 100;
      $timeout(function() {
        angular.element('body').css('overflow', '');
        $animate.addClass(el, 'preloader-hidden');
      }, 300);
    }

    function appReady() {
      var deferred = $q.defer();
      var viewsLoaded = 0;
      // TODO: Refactor with custom event if doesn't work
      var off = scope.$on('$viewContentLoaded', function() {
        viewsLoaded++;
        if (viewsLoaded === 2) {  // with resolve this fires only once
          $timeout(function() {
            deferred.resolve();
          }, 3000);
          off();
        }
      });
      return deferred.promise;
    }
  }
}
```
